src/cljc/matcher_combinators/core.cljc

```diff
@@ -143,6 +143,16 @@
       issue
       (compare-maps expected actual model/->Unexpected false))))
 
+(defrecord EqualsRecord [expected]
+  Matcher
+  (match [_this actual]
+    (if-let [issue (validate-input expected actual map? 'equals "record")]
```

Comment:
I think you should use the other arity of `validate-input` and do `(validate-input expected actual record? map? 'equals "record")`